### PR TITLE
docs: various fixes for tf.keras docs.

### PR DESCRIPTION
**Pull request by neilconway**
* Clarify docs on epoch boundaries
* Fix epoch size in Fashion MNIST example
* Make wrapper requirements more clear
* Better linking to API reference.

## Commentary (optional)

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: docs/tutorials/tf-mnist-tutorial.txt [hunk 1]
```diff
@@ -129,25 +129,25 @@ For more information on loading data in Determined, refer to the tutorial on :re
...
+    records_per_epoch: 50000
     searcher:
         name: single
         metric: val_accuracy
         max_length:
           epochs: 5
     entrypoint: model_def:FashionMNISTTrial
 
-For this model, we have two hyperparameters: the size of the ``Dense`` layer and the batch size. We train the model on five epochs and should reach about 85% accuracy on the validation set, which mimics the original ``tf.keras`` implementation.
+For this model, we have chosen two hyperparameters: the size of the ``Dense`` layer and the batch size. Training the model on five epochs of data should reach about 85% accuracy on the validation set, which mimics the original ``tf.keras`` implementation.
```
Comment (stoksc) [inline docs/tutorials/tf-mnist-tutorial.txt:150]: Maybe "Training the model for five epochs"?

File: examples/official/trial/mnist_pytorch/distributed.yaml [hunk 2]
```diff
@@ -9,8 +9,7 @@ hyperparameters:
   dropout1: 0.25
   dropout2: 0.5
 resources:
-  # Use 16 GPUs to train the model.
-  slots_per_trial: 16
+  slots_per_trial: 8
```
Comment (stoksc) [inline examples/official/trial/mnist_pytorch/distributed.yaml:12]: What's the reason for this change? Just curious, seems reasonable because this is what the tests run anyways (`test_pytorch_parallel`).

File: harness/determined/keras/_tf_keras_trial.py [hunk 3]
```diff
@@ -713,9 +732,12 @@ def keras_callbacks(self) -> List[tf.keras.callbacks.Callback]:
         Determined training behavior.
 
         .. note::
-            If a callback is supplied that has implemented `keras.callbacks.Callback.on_epoch_end
+            If you specify a Keras callback that uses the `on_epoch_begin
+            <https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/Callback#on_epoch_begin>`__
```
Comment (stoksc) [inline harness/determined/keras/_tf_keras_trial.py:736]: ah, good catch adding `on_epoch_begin`, too.

File: examples/official/trial/mnist_pytorch/distributed.yaml (hunk 2 above)
Comment (neilconway) [reply examples/official/trial/mnist_pytorch/distributed.yaml:12]: The quick start tutorial references this file but claims that `slots_per_trial` is 8. Seems like a reasonable default value anyway...
